Controller/Config/Fast.php

```diff
@@ -100,54 +100,82 @@ public function __construct(
+            'cartId' => '',
+            'theme' => $theme
+        ];
+
+        try {
+            $quote = $this->checkoutSession->getQuote();
+            $cartId = $this->getInvisibleCartFromCart($quote);
+            $data['cartId'] = $cartId;
+
+            if(!$quote || !$quote->getId()){
+                return $result->setData($data);
             }
-            if ($cartItem->getProductType() === 'bundle' ||
-                $cartItem->getProductType() === 'downloadable') {
-                $cartIsFast = false;
+
+            $cartIsFast = true;
+            foreach ($quote->getAllVisibleItems() as $cartItem) {
+                if ((int)$cartItem->getProduct()->getData('hide_fast_option') == 0) {
+                    $cartIsFast = false;
+                }
+                if ($cartItem->getProductType() === 'bundle' ||
+                    $cartItem->getProductType() === 'downloadable') {
+                    $cartIsFast = false;
+                }
             }
+            $data['areAllProductsFast'] = $cartIsFast;
+        } catch (\Exception $exception) {
+            $this->fastCheckoutHelper->log('No quote found.');
         }
-        return $result->setData([
-            'areAllProductsFast' => $cartIsFast,
-            'success' => true,
-            'appId' => $this->fastConfig->getAppId(),
-            'cartId' => $cartId,
-            'theme' => $this->fastConfig->useDarkTheme() ? 'dark' : ''
-        ]);
+
+        $data['success'] = true;
+        return $result->setData($data);
     }
 
-    public function getInvisibleCartFromCart(Quote $quote): string
+    /**
+     * Get Invisible cart
+     *
+     * @param Quote|null $quote
+     * @return string - cartId
+     *
+     */
+    public function getInvisibleCartFromCart(?Quote $quote): string
     {
         //Get cart from cart ID
-        /** @var Quote $currentCart */
-        $currentCart = $this->quoteRepository->get($quote->getEntityId());
         $maskedId = $this->cartManagement->createEmptyCart();
         $this->fastCheckoutHelper->log($maskedId);
         $newCartId = $this->maskedQuoteIdToQuoteId->execute($maskedId);
         /** @var Quote $newCart */
         $newCart = $this->quoteFactory->create()->load($newCartId, 'entity_id');
-        //$newCart->setData($currentCart->getData());
         $newCart->setActive(0);
-        $newCart->setCouponCode($currentCart->getCouponCode());
-        //$newCart->removeAllItems();
-        foreach ($currentCart->getAllVisibleItems() as $item) {
-            $this->fastCheckoutHelper->log('item ' . $item->getName());
-            $newItem = clone $item;
-            $newCart->addItem($newItem);
-            if ($item->getHasChildren()) {
-                foreach ($item->getChildren() as $child) {
-                    $newChild = clone $child;
-                    $newChild->setParentItem($newItem);
-                    $newCart->addItem($newChild);
+
+        if ($quote && $quote->getId()) {
```

Comment:
Should this be `if ($quote && $quote->getEntityId()) {`?

Reply:
@corywebb This should be equal to getEntityId(). 
getId method get the value from `$_idFieldName` which gets set to `entity_id` in quote collection class.

https://www.magentoextensions.org/documentation/class_magento_1_1_quote_1_1_model_1_1_resource_model_1_1_quote.html#aa1086f6e600afd7f5d30f22fccf0effc